### PR TITLE
[csrng/dv] Clean up CSRNG fips coverage

### PR DESCRIPTION
**Pull request by h-filali**
This PR cleans up the code that makes up the fips coverage.
For a more detailed explanation please see the individual commit messages.

Most of this PR is already included in #20475.

### CODE REVIEW

**Review by vogelpi — COMMENTED** [review]
Thanks for working on this @h-filali . The implementation looks good. As discussed in person, we should also add transition coverage 1) for the fips bit received from the entropy_src agent and 2) for the actual fips bit of the individual instances. Because the fips bit of the instance depends on the es input and flag0. We would like to see all these transitions and be sure that CSRNG correctly propagates them through to the EDN / SW outputs.

**Comment by vogelpi** [comment]
I've marked this as V3, we will not merge this during the quiet period.

File: hw/ip/csrng/dv/cov/csrng_cov_if.sv
Comment (h-filali) [reply hw/ip/csrng/dv/cov/csrng_cov_if.sv:469]: This represents a range for the bin. AFAIK we don't have a guideline in the style guide for this. And the duolos manual suggests to write it this way.

File: hw/ip/csrng/dv/cov/csrng_cov_if.sv
Comment (h-filali) [reply hw/ip/csrng/dv/cov/csrng_cov_if.sv:472]: Thanks for the suggestion, I'll change it to make it more clear.

File: hw/ip/csrng/dv/cov/csrng_cov_if.sv
Comment (h-filali) [reply hw/ip/csrng/dv/cov/csrng_cov_if.sv]: I'll add this to the commit message.

**Review by h-filali — COMMENTED** [review]
Thanks for the review @vogelpi

**Review by vogelpi — APPROVED** [review]
LGTM, thanks @h-filali ! Once #20475, you'll have to rebase this.

**Comment by vogelpi** [comment]
> LGTM, thanks @h-filali ! Once #20475, you'll have to rebase this.

@h-filali , #20475 has been merged. If you can rebase this one, I will merge it later tonight.

**Comment by vogelpi** [comment]
I've updated this PR on master and fixed a missing argument in the call to the genbits covergroup sampling function.

**Pull request merged**
